catalog/services_state.go

```diff
@@ -230,6 +230,23 @@ func (state *ServicesState) AddListener(listener Listener) {
 	log.Debugf("AddListener(): added %s, new count %d", listener.Name(), len(state.listeners))
 }
 
+// Remove an event listener channel by name. This will find the first
+// listener in the list with the specified name and will remove it.
+func (state *ServicesState) RemoveListener(name string) error {
+	state.Lock()
+	defer state.Unlock()
+
+	for i := 0; i < len(state.listeners); i++ {
+		if state.listeners[i].Name() == name {
+			state.listeners = append(state.listeners[:i], state.listeners[i+1:]...)
```

Comment:
Yeah, maybe, but that's not as easy to understand and performance is pretty irrelevant here. The one-liner I used is the kind of normal way to remove an entry from a slice.